### PR TITLE
Sort detections in candidate page by absolute timestamp, low to high

### LINKED ISSUE #278: Report view of a candidate numbers detections out of chronological order
**Describe the bug**
As moderator reviewing detections and candidates, it would be ideal if the playback UI for a candidate numbered detections chronologically by detection `timestamp`, rather than by the unique internal `ID` field.

**To Reproduce**

1. Go to [candidate cand_02up4cYJD0vJ6aqbAphhg7](https://live.orcasound.net/reports/cand_02up4cYJD0vJ6aqbAphhg7) with 7 detections
2. Notice that in the playback UI (recording timeline) and the detection table, the `#` is determined by the alphanumeric order of the unique internal `ID` field (e.g. det_02up4`c`YIQeYBt8cWTtUNYc comes first before det_02up4`e`lhztRr3KlMrdVwxz because `c` comes before `e` in the alphabet).

**Ideal behavior**
Ideally, the numbering of detctions in both the playback UI and the table would be determined by the `Timestamp` value, with earliest detections being numbered first.

**Screenshots**

Screenshot 2023-11-14 at 12 31 02 PM

**Desktop (please complete the following information):**
 - OS: OSX
 - Browser: Chrome

**Additional idea**
It it makes sorting easier, I have a slight preference for 24-hour time in the formatting of the `Timestamp` field in this UI, rather than the current AM/PM formatting.

### PR DESCRIPTION
**Pull request by skanderm**
- Closes #278

### CODE REVIEW

File: ui/src/components/DetectionsTable.tsx [hunk 1]
```diff
@@ -111,7 +111,10 @@ export default function DetectionsTable({
         <TableBody>
           {detections
             .slice()
-            .sort((a, b) => a.id.localeCompare(b.id))
+            .sort(({ timestamp: a }, { timestamp: b }) =>
+              // Sort by timestamp, low to high
+              a == b ? 0 : a < b ? -1 : 0,
```
Comment (paulcretu) [inline ui/src/components/DetectionsTable.tsx:116]:
```suggestion
              a === b ? 0 : a < b ? -1 : 0,
```

Comment (paulcretu) [reply ui/src/components/DetectionsTable.tsx:116]: But also, are these `string` or `Date`? Typescript say they should be `Date` but I don't think we ever deserialize, they look like ISO 8601 strings to me. I think we should figure out deserialization, it's safer and then you could just subtract to sort.

Don't want to bog down your PR though, can do it as a separate issue

**Review by paulcretu — APPROVED** [review]

File: ui/src/components/DetectionsTable.tsx (hunk 1 above)
Comment (paulcretu) [reply ui/src/components/DetectionsTable.tsx:116]: Actually, how about just manually doing `new Date()` for now, until we get deserialization working?

**Review by paulcretu — CHANGES_REQUESTED** [review]

File: ui/src/components/DetectionsTable.tsx (hunk 1 above)
Comment (skanderm) [reply ui/src/components/DetectionsTable.tsx]: Done!

File: ui/src/components/DetectionsTable.tsx [hunk 2]
```diff
@@ -111,7 +111,13 @@ export default function DetectionsTable({
         <TableBody>
           {detections
             .slice()
-            .sort((a, b) => a.id.localeCompare(b.id))
+            .sort(({ timestamp: a }, { timestamp: b }) => {
+              const date_a = new Date(a);
+              const date_b = new Date(b);
+
+              // Sort by timestamp, low to high
+              return date_a === date_b ? 0 : date_a < date_b ? -1 : 0;
```
Comment (paulcretu) [inline ui/src/components/DetectionsTable.tsx:119]:
```suggestion
              return +date_a - +date_b;
```

**Review by paulcretu — CHANGES_REQUESTED** [review]

**Review by paulcretu — APPROVED** [review]

**Pull request merged**